cli/spec/spec_helper.rb

```diff
@@ -13,43 +13,54 @@
 # it.
 #
 # See https://rubydoc.info/gems/rspec-core/RSpec/Core/Configuration
-ENV["RACK_ENV"] ||= "test"
-
 require "./cli"
+require "dotenv"
+require "rom-factory"
+require "database_cleaner/sequel"
 
-module Lauth
-  module API
-    class BDD
-      def self.rom
-        @@rom ||= ::ROM.container(:sql, "mysql2:///#{CONFIG["db"]["name"]}", CONFIG["db"]) do |config|
-          config.auto_registration("../lib/lauth/api/rom", namespace: "Lauth::API::ROM")
-        end
-      end
-    end
-  end
-end
+Dotenv.load
 
-require "rom-factory"
+MYSQL_DB = {
+  adapter: ENV["LAUTH_API_DB_ADAPTER"] || :mysql2,
+  host: ENV["LAUTH_API_DB_HOST"] || "localhost",
+  port: ENV["LAUTH_API_DB_PORT"] || 3306,
+  database: ENV["LAUTH_API_DB_DATABASE"] || "test",
+  user: ENV["LAUTH_API_DB_USER"] || "root",
+  password: ENV["LAUTH_API_DB_PASSWORD"] || "",
+  encoding: ENV["LAUTH_API_DB_ENCODING"] || "utf8mb4"
+}.freeze
+
+MYSQL_CMD = "mysql --user=#{MYSQL_DB[:user]} --host=#{MYSQL_DB[:host]} --port=#{MYSQL_DB[:port]} --password=#{MYSQL_DB[:password]}  #{MYSQL_DB[:database]}".freeze
+
+puts MYSQL_DB.to_s
+puts MYSQL_CMD.to_s
+
+LAUTH_API_ROM = ::ROM.container(:sql, MYSQL_DB) do |config|
+  config.auto_registration("../lib/lauth/api/rom", namespace: "Lauth::API::ROM")
+end
 
 Factory = ROM::Factory.configure do |config|
-  config.rom = Lauth::API::BDD.rom
+  config.rom = LAUTH_API_ROM
 end
 
 factories_dir = File.expand_path("../../../lib/lauth/api/factories", __FILE__)
 Dir[factories_dir + "/*.rb"].sort.each { |file| require file }
 
-require "database_cleaner/sequel"
-
 RSpec.configure do |config|
   config.before(:suite) do
-    DatabaseCleaner.strategy = :transaction
-    DatabaseCleaner.clean_with(:truncation)
+    # DatabaseCleaner.strategy = :transaction
+    DatabaseCleaner.strategy = :truncation
```

Comment:
To confirm my understanding: we use transaction in the API tests because it "owns" the database and truncation from the CLI tests because driving the API will do things that CLI a transaction won't cover?

Reply:
Correct.